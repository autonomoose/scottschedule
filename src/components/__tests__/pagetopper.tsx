import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

import PageTopper from "../pagetopper";
//                <Link key="home" to={linkDebug('/home', vdebug)}>Home</Link>
//              <Link key={item.title}
//                to={linkDebug(item.plink, vdebug)}>
//                {item.title}</Link>
//             <Link to={linkDebug(item.plink,vdebug)}>
//               <small>{item.title}</small></Link>



const myhelp = '/help/pagetopper';
const myLinks = [
    {title: 'bcrumb1', plink: `/bcrumblink1`},
    {title: 'bcrumb2', plink: `/bcrumblink2?test=special`},
    ];
const myXtras = [
    {title: 'xtra1', plink: `/xtralink1`},
    {title: 'xtra2', plink: `/xtralink2`},
    ];

describe("PageTopper", () => {
    it("renders home snapshot correctly", async () => {
      const {container} = render(
        <PageTopper pname='Home' ptitle='Testing Home'
          helpPage={myhelp}
          linksList={myLinks}
          xtraList={myXtras}
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    it("renders debug", async () => {
      render(
        <PageTopper pname='Home' vdebug='true'
        helpPage={myhelp}
        linksList={myLinks}
        xtraList={myXtras}
        />
      );
    });
    it("renders /wo title or help", async () => {
      render(
        <PageTopper pname='Home'
        linksList={myLinks}
        xtraList={myXtras}
        />
      );
    });
});


