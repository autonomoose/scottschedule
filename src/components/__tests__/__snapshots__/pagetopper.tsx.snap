// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTopper renders home snapshot correctly 1`] = `
<div
  class="MuiBox-root css-7kebnq"
>
  <div
    style="float: left;"
  >
    <nav
      aria-label="Breadcrumb"
      class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
    >
      <ol
        class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
            href="/bcrumblink1"
          >
            bcrumb1
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
        >
          &gt;
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
            href="/bcrumblink2?test=special"
          >
            bcrumb2
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
        >
          &gt;
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <span>
            Home
          </span>
        </li>
      </ol>
    </nav>
    <h2>
      Testing Home
    </h2>
  </div>
  <div
    class="MuiBox-root css-0"
  >
    <button
      class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-1k23hlb-MuiButtonBase-root-MuiButton-root"
      style="margin: 0px 2px;"
      tabindex="0"
      type="button"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
        href="/xtralink1"
      >
        <small>
          xtra1
        </small>
      </a>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-1k23hlb-MuiButtonBase-root-MuiButton-root"
      style="margin: 0px 2px;"
      tabindex="0"
      type="button"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
        href="/xtralink2"
      >
        <small>
          xtra2
        </small>
      </a>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div>
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1kt0v4n-MuiTypography-root-MuiLink-root"
      href="/help/pagetopper"
    >
      Help
    </a>
  </div>
</div>
`;
